sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -484,7 +557,7 @@ public EventHubConsumerClient buildConsumerClient() {
      *     either {@link #connectionString(String)} or {@link #credential(String, String, TokenCredential)}. Or, if a
      *     proxy is specified but the transport type is not {@link AmqpTransportType#AMQP_WEB_SOCKETS web sockets}.
      */
-    public EventHubProducerAsyncClient buildAsyncProducerClient() {
+    public EventHubProducerClient buildAsyncProducerClient() {
```

Comment:
This should return EventHubProducerAsyncClient.

Reply:
According to your comment in the new revision amendments have been made.